components/codec/src/byte.rs

```diff
@@ -331,7 +339,7 @@ impl MemComparableByteCodec {
     ///
     /// Please refer to `try_decode_first` for the meaning of return values, panics and errors.
     #[inline]
-    fn try_decode_first_internal<T: MemComparableCodecHelper>(
+    unsafe fn try_decode_first_internal<T: MemComparableCodecHelper>(
```

Comment:
Could you document in the doc comment why the function is unsafe and what the caller must do to keep things safe please?

Reply:
Done.